arrow-fx-coroutines/README.MD

```diff
@@ -0,0 +1,480 @@
+    .apply { isDaemon = true}
+  }
+}
+
+suspend fun shutdown(service: ExecutorService): Unit =
+  service.shutdown()
+
+suspend fun singleThreadContext(name: String): Resource<CoroutineContext> =
+  Resource({ service(1, name) }, ::shutdown)
+    .map(ScheduledExecutorService::asCoroutineContext)
+
+suspend fun main(): Unit {
+  singleThreadContext("example-pool").use { ctx ->
+     evalOn(ctx) {
+        println("Running on example-pool: $Thread.currenThread().name")
+     }
+  }
+}
+```
+
+A more advanced `Resource` example that reads 3 `DatabaseConnection`s in parallel
+
+```kotlin:ank
+class File(url: String) {
+  fun open(): File = this
+  suspend fun close(): Unit {}
+  override fun toString(): String = "This file contains some interesting content!"
+}
+
+suspend fun openFile(uri: String): File = File(uri).open()
+
+val resources: List<Resource<File>> =
+ listOf(
+   Resource({ openFile("path1") }, File::close),
+   Resource({ openFile("path2") }, File::close),
+   Resource({ openFile("path3") }, File::close)
+ )
+
+val resource: Resource<List<File>> =
+  resources.sequence(Resource.applicative())
+
+suspend main(): Unit {
+  resource.use { files ->
+    files.parTraverse(IODispatchers.IOPool) { file ->
+       file.toString()
+    }
+  }
+}
+```
+
+### Error Handling
+
+In Kotlin with suspend `try/catch` can safely be used to recover from exceptions.
+
+Simple constructs like `suspend fun Either.catch(f: () -> A): Either<Throwable, A>` are available for nicer syntax when and lifting errors in your domain.
+
+### Retrying and repeating effects
+
+`Schedule` allows you to define and compose powerful yet simple policies, which can be used to either repeat or retry computation. 
+
+A simple example might be to repeat an action `n` times, similar to the `repeat` function in the standard library.
+
+```kotlin:ank
+suspend main(): Unit {
+  repeat(Schedule.recurs<A>(n)) {
+    println("Hello")
+  }
+}
+```
+
+Alternatively we can re-use this `Schedule` to `retry` a `suspend fun` `n` times when it fails.
+
+```kotlin:ank
+suspend main(): Unit {
+  retry(Schedule.recurs<A>(n)) {
+    println("I am going to do nothing but throw a tantrum!")
+    throw RuntimeException("Boom!")
+  }
+}
+```
+
+Additionally a `Schedule` can also produce an output, let's compose our simple `recurs` Schedule to `recur` every `10.seconds` in a `spaced` fashion and collect all outputs along the way.
+
+```kotlin:ank
+fun <A> schedule(): Schedule<A, List<A>> = Schedule {
+  (recurs<A>(10) and spaced(10.seconds)) zipRight collect()
+}
+
+suspend main(): Unit {
+  var count = Atomic(0)
+
+  val history: List<Int> = repeat(schedule<Int>()) {
+    println("Incrementing the ref")
+    count.update(Int::inc)
+  }
+}
+```
+
+## Parallel Ops
+
+
+### ForkConnected
+
+`forkConnected` launches a `suspend () -> A` function in a seperate processes called a `Fiber`, the lifecycle of this process is connected to its parent.
```

Comment:
*separate

Process? -> Thread/Coroutine/Executor?

Reply:
forkConnected` launches a `suspend () -> A` function on a given `CoroutineContext` inside a `Fiber`

?

Reply:
`forkConnected` launches a `suspend () -> A` function inside a `Fiber` on a given `CoroutineContext`. The lifecycle of this `Fiber` is connected to its parent.